### PR TITLE
allow to pass custom event on brush start

### PR DESCRIPTION
**Pull request by bartes**
POC - but this was required to handle some synced drawing across multiple charts

- added ability to pass custom event on brush.start
- added ability to pass alternative (named `desiredTarget`) to brush.move 
- temporary added dist to repo for testing purposes*

Is there a possibility to introduce those functionality with this shape or differently

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,4 @@
 *.sublime-workspace
 .DS_Store
-dist/
```
Comment (curran) [inline .gitignore]: Umm this should definitely remain in `.gitignore`.

File: dist/d3-brush.js [hunk 2]
```diff
@@ -0,0 +1,624 @@
+// https://d3js.org/d3-brush/ v1.1.5 Copyright 2020 Mike Bostock
```
Comment (curran) [inline dist/d3-brush.js]: This file should not be here.

File: src/brush.js [hunk 3]
```diff
@@ -215,6 +215,13 @@ function brush(dim) {
         .style("-webkit-tap-highlight-color", "rgba(0,0,0,0)");
   }
 
+  brush.start = function(group, passedEvent) {
```
Comment (curran) [inline src/brush.js]: Without an example, it's not clear why this change is needed.

**Comment by mbostock** [comment]
Likewise, I don’t understand the motivation for this change, and this PR includes a number of unrelated changes that are also undesirable.

**Comment by Fil** [comment]
[*brush*.move](https://github.com/d3/d3-brush#brush_move) allows syncing between charts. See https://observablehq.com/@d3/brushable-scatterplot-matrix#brush for an example.

**Pull request closed without merging**